core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/MSYS2.java

```diff
@@ -0,0 +1,164 @@
+/********************************************************************************
+ * Copyright (c) 2022 徐持恒 Xu Chiheng
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.cdt.utils;
+
+import java.io.File;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Platform;
+
+// A collection of MSYS2-related utilities.
+public class MSYS2 {
+	public static boolean present;
+	private static String rootDir;
+	static {
+		initialize();
+	}
+
+	// initialize static data fields
+	private static void initialize() {
+		if (Platform.getOS().equals(Platform.OS_WIN32)) {
+			Map<String, String> environment = System.getenv();
+			rootDir = environment.get("MSYS2_DIR"); //$NON-NLS-1$
+			if (rootDir != null) {
+				if (dirHasMsys2Dll(rootDir)) {
+					present = true;
+					rootDir = new Path(rootDir).toPortableString();
```

Comment:
toPortableString is only for use when you are later calling fromPortableString. See [javadoc](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/runtime/IPath.html#toPortableString()) for more info, key part quoted here for convenience:

> Returns a platform-neutral string representation of this path. The format is not specified, except that the resulting string can be passed back to the Path#fromPortableString(String) constructor to produce the exact same path on any platform.

I think you want use [toString](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/runtime/IPath.html#toString()) which is explicitly documented as using `/` which is my assumption as to why you are using toPortableString:

> Returns a string representation of this path, including its device id. The same separator, "/", is used on all platforms.